docker.go

```diff
@@ -690,6 +694,29 @@ func (c *DockerContainer) copyToContainer(ctx context.Context, fileContent func(
 	return nil
 }
 
+// logConsumerWriter is a writer that writes to a LogConsumer.
+type logConsumerWriter struct {
+	log       Log
+	consumers []LogConsumer
```

Comment:
Offtopic: In the v1 branch I have a LogConsumer that is a Multi log consumer (https://github.com/testcontainers/testcontainers-go/commit/9de581c05a541927a825c1b304e22ca818e4178f#diff-7a89ef1d0fc02a6398aad1d81b64b68f375822b4de5b15458f812daacf93595dR21), which would simplify this. I did not backport it from v1 to main to avoid another breaking change. Although it could be convenient if we see it useful.

Reply:
We should consider eliminating the current Accept for something more idiomatic, separate readers for stdout and stderr, then you can just use [io.MultiReader](https://pkg.go.dev/io#MultiReader)